Sources/KeyboardShortcuts/SwiftUI+Extensions.swift

```diff
@@ -0,0 +1,123 @@
+// Created by Neil Clayton on 17/07/2024.
+// Using ideas by mbenoukaiss, https://github.com/sindresorhus/KeyboardShortcuts/issues/101
+
+
+import Foundation
+import SwiftUI
+import KeyboardShortcuts
+import Carbon
+
+// Provides a SwiftUI like wrapper func, that feels the same as the normal SwiftUI keyboardShortcut
+@available(macOS 11.0, *)
+extension View {
+	@ViewBuilder
+	public func keyboardShortcut(_ shortcutName: KeyboardShortcuts.Name) -> some View {
+		KeyboardShortcutView(shortcutName: shortcutName) {
+			self
+		}
+	}
+}
+
+// Holds the state of the shortcut, and changes that state when the shortcut changes
+// This causes the related NSMenuItem to also update (yipeee)
+@available(macOS 11.0, *)
+struct KeyboardShortcutView<Content: View>: View {
+	@State private var shortcutName: KeyboardShortcuts.Name
+	@State private var shortcut: KeyboardShortcuts.Shortcut?
+
+	private var content: () -> Content
+
+	init(shortcutName: KeyboardShortcuts.Name, content: @escaping () -> Content) {
+		self.shortcutName = shortcutName
+		self.shortcut = KeyboardShortcuts.getShortcut(for: shortcutName)
+		self.content = content
+	}
+
+	@ViewBuilder
+	var shortcutBody: some View {
+		if let shortcut, let keyEquivalent = shortcut.toKeyEquivalent() {
```

Comment:
You don't need to do `if` and ruin the view `id` when it changes. Instead, use https://developer.apple.com/documentation/financekitui/addordertowalletbutton/keyboardshortcut(_:)-883ia#